### ISSUE TITLE
Allow using RSwag on specs other than :request

### ISSUE THREAD

**Issue by hiddewie** [rswag/rswag#364]
There is a fixed `:request` spec extension configured in https://github.com/rswag/rswag/blob/master/rswag-specs/lib/rswag/specs.rb.

This extension pollutes the namespace of other `:request` specs that do not use the RSwag DSL. 

Can the type of specs that the RSwag spec DSL is registered to, be made configurable?

For example, I would like to specify `:openapi` specs to use the RSwag spec DSL, run the examples and generate the OpenAPI specifications, but not the `:request` specs.

**Comment by hiddewie** [comment]
Not stale!

**Comment by BookOfGreg** [comment]
I agree, this is not ideal for us to hard-code the name we're locked onto. Do you see a nice way of pulling it into the rswag config? I'd happily accept a PR in this area if you could contribute the time? :pray:

**Comment by hiddewie** [comment]
There is a problem that the RSpec extension is configured before the user can define the config with `RSpec.configure { }`

I could image something like 
```diff
diff --git rswag-specs/lib/rswag/specs.rb rswag-specs/lib/rswag/specs.rb
index 1db62a5..1c605ea 100644
--- rswag-specs/lib/rswag/specs.rb
+++ rswag-specs/lib/rswag/specs.rb
@@ -10,12 +10,15 @@ module Rswag
   module Specs
     # Extend RSpec with a swagger-based DSL
     ::RSpec.configure do |c|
+
+      extension_type = defined?(Rswag::RSPEC_TYPE) ? Rswag::RSPEC_TYPE : :request
+
       c.add_setting :swagger_root
       c.add_setting :swagger_docs
       c.add_setting :swagger_dry_run
       c.add_setting :swagger_format
-      c.extend ExampleGroupHelpers, type: :request
-      c.include ExampleHelpers, type: :request
+      c.extend ExampleGroupHelpers, type: extension_type
+      c.include ExampleHelpers, type: extension_type
     end

     def self.config
```

But the user would need to define `Rswag::RSPEC_TYPE` before the rswag code is included. Not sure if that is reasonable. Otherwise this issue is hard to solve.

**Comment by hiddewie** [comment]
Not stale!

**Comment by romanblanco** [comment]
@jtannas, can you reopen this issue?

**Comment by hiddewie** [comment]
Not stale!

**Comment by atruskie** [comment]
One upping so stale bot won't attack again.

I have my rawag tests in `spec/api` and more complex integration tests that aren't suitable for documentation in `spec/request`.

When I went to use `example` (and alias for `it`) in my non-rswag folder of tests I was hit with a confusing error about not providing enough arguments. When I looked to configure the scope for rswag methods were applied, I found this issue.

**Comment by hiddewie** [comment]
Not stale

**Comment by hiddewie** [comment]
Not stale
